backend/src/layer/data/database/request.rs

```diff
@@ -286,6 +286,47 @@ impl RequestDataAccess for PersistentRequestData {
         .await?;
         Ok(res)
     }
+
+    async fn get_nutrition_data(&self, food_id: Uuid) -> Result<Option<NutritionData>> {
+        let res = sqlx::query!(
+            r#"SELECT energy, protein, carbohydrates, sugar, fat, saturated_fat, salt FROM food_nutrition_data WHERE food_id = $1"#,
+            food_id
+        ).fetch_optional(&self.pool)
+        .await?;
+        if res.is_none() {
+            return Ok(None);
+        }
+        let res = res.expect("isdpoiopvü");
```

Comment:
Maybe something like
```rust
let res = match res {
  Some(res) => res,
  None => return Ok(None)
}
```
Or you could use Option::map but the closure would need to return a Result, so you get Option<Result<...>>. On that, you can call Option::transpose [1] to get Result<Option<..>,..>. I think that is what we want, maybe there is also an early return version using transpose and ?, but things get complicated here, so the match may be the most readable.

[1] https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose